core/input/input_map.cpp

```diff
@@ -616,6 +618,16 @@ const HashMap<String, List<Ref<InputEvent>>> &InputMap::get_builtins() {
 	inputs.push_back(InputEventKey::create_reference(Key::DOWN | KeyModifierMask::CMD_OR_CTRL));
 	default_builtin_cache.insert("ui_text_caret_document_end.macos", inputs);
 
+	// Text Caret Insertion Below/Above
+
+	inputs = List<Ref<InputEvent>>();
+	inputs.push_back(InputEventKey::create_reference(Key::DOWN | KeyModifierMask::SHIFT | KeyModifierMask::CMD_OR_CTRL));
```

Comment:
<kbd>Command/Ctrl</kbd> + <kbd>Down/Up</kbd> are already mapped to `ui_text_scroll_down/up`, so these shortcuts will conflict with existing "select to the end/beginning" of text behavior.

Reply:
Correct me if I am wrong, but I think this is what the `p_exact_match` parameter of `InputEvent::is_action` is taking care of. It leads to the method only returning true if the input keys are exactly that of the action, which would result in these two actions not conflicting.

Reply:
I just tested it and they are not conflicting.

Reply:
It is confliction, but only on macOS (with `ui_text_caret_document_end/start`), selection to the end/beginning of the document is no longer possible.

Reply:
You are right. If you would press `Ctrl+Shift+Up` in the intentation of selecting everything from here to the top of the document it would conflict. I will change the default buttons. Any suggestions?

Reply:
I'm not sure if there are any free options to use.

Reply:
I changed it to `Cmd/Ctrl+Shift+O` and `Cmd/Ctrl+Shift+L`